File: generate_icons.js
import fs from 'fs';
import path from 'path';

const transparentPngBase64 = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=';
const buffer = Buffer.from(transparentPngBase64, 'base64');

const densities = ['mipmap-mdpi', 'mipmap-hdpi', 'mipmap-xhdpi', 'mipmap-xxhdpi', 'mipmap-xxxhdpi'];
const fileNames = ['ic_launcher.png', 'ic_launcher_round.png', 'ic_launcher_foreground.png'];

densities.forEach(density => {
  const dir = path.join('android', 'app', 'src', 'main', 'res', density);
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }
  fileNames.forEach(fileName => {
    const filePath = path.join(dir, fileName);
    fs.writeFileSync(filePath, buffer);
    console.log(`Generated valid PNG: ${filePath}`);
  });
});
console.log('All launcher icons successfully regenerated as valid PNGs!');

File: app/applet/generate_icons.py
import os
import struct
import zlib

def make_png(width, height, color=(37, 99, 235, 255)):
    row = b'\x00' + bytes(color * width)
    raw_data = row * height
    compressed_data = zlib.compress(raw_data)
    
    def chunk(tag, data):
        return struct.pack('>I', len(data)) + tag + data + struct.pack('>I', zlib.crc32(tag + data) & 0xffffffff)

    ihdr = struct.pack('>IIBBBBB', width, height, 8, 6, 0, 0, 0)
    return b'\x89PNG\r\n\x1a\n' + chunk(b'IHDR', ihdr) + chunk(b'IDAT', compressed_data) + chunk(b'IEND', b'')

densities = {
    'mipmap-mdpi': 48,
    'mipmap-hdpi': 72,
    'mipmap-xhdpi': 96,
    'mipmap-xxhdpi': 144,
    'mipmap-xxxhdpi': 192
}

base_dir = 'android/app/src/main/res'
for folder, size in densities.items():
    dir_path = os.path.join(base_dir, folder)
    os.makedirs(dir_path, exist_ok=True)
    
    # Generate ic_launcher.png (Solid blue with white center or clean blue)
    png_data = make_png(size, size, (37, 99, 235, 255))
    with open(os.path.join(dir_path, 'ic_launcher.png'), 'wb') as f:
        f.write(png_data)
        
    # Generate ic_launcher_round.png
    with open(os.path.join(dir_path, 'ic_launcher_round.png'), 'wb') as f:
        f.write(png_data)
        
    # Generate ic_launcher_foreground.png
    with open(os.path.join(dir_path, 'ic_launcher_foreground.png'), 'wb') as f:
        f.write(png_data)

print("All launcher icons successfully generated!")
